src/dispatch/stage.rs

```diff
@@ -242,20 +242,45 @@ impl<'a> StagesBuilder<'a> {
         let new_writes = new_writes.into_iter();
 
         let num_groups = ids[stage].len();
+        let mut dep_conflict = false;
 
         let conflict = (0..num_groups)
             .filter(|&group| {
-                let read_and_write = writes[stage][group]
+                let reads_and_writes = writes[stage][group]
                     .iter()
                     .chain(reads[stage][group].iter());
 
-                check_intersection(new_writes.clone(), read_and_write) ||
-                check_intersection(new_reads.clone(), writes[stage][group].iter()) ||
-                check_intersection(new_dep.iter(), ids[stage][group].iter())
+                let inters = check_intersection(new_writes.clone(), reads_and_writes) ||
+                check_intersection(new_reads.clone(), writes[stage][group].iter());
+
+                if inters {
+                    true
+                } else {
+                    if check_intersection(new_dep.iter(), ids[stage][group].iter()) {
+                        dep_conflict = true;
+
+                        true
+                    } else {
+                        false
+                    }
+                }
+
             })
             .fold(Conflict::None, Conflict::add);
 
-        conflict
+        // If there is a dependency in the dependency list
+        // which was not in a previous or this stage,
+        // return `Multiple` conflict.
+
+        // true if there's at least one unmet dependency
+        let unmet_dependency = !new_dep.is_empty();
+
+
+        match conflict {
+            Conflict::None if unmet_dependency => Conflict::Multiple,
+            Conflict::Single(_) if (dep_conflict && new_dep.len() > 1) || unmet_dependency => Conflict::Multiple,
```

Comment:
This still has the same problem.
The shape of the guard is:
````rust
let a: bool;
let n: usize;
if (a && n > 1) || n > 0 {
    // Multiple conflicts
}
````
and so `a`s value is irrelevant, because `n > 0` is true for all values of `n` for which `n > 1` is true.

Reply:
Oops, you're right.